data_structures/k_tree/k_tree.py

```diff
@@ -0,0 +1,79 @@
+class Node:
+    """defines each node used to build the k-ary tree"""
+    def __init__(self, val, input_list=[]):
+        self.val = val
+        self.children = []
+        for child in input_list:
+            self.children.append(Node(child))
+
+    def __repr__(self):
+        # return 'node children: {} '.format(lambda x: for x in self.children: return x.val)
```

Comment:
If you want to do something like this, you need to use a list comprehension instead.
```python
.format([x.val for x in self.children])
```